File: cool-load/src/model/Container.js
export default class Container{
    constructor(client,id,type,status,category){
        this.client = client;
        this.id = id;
        this.type = type;
        this.status = status;
        this.category = category;
    }
}

export class TypeContainer{
    static T20 = new TypeContainer("T20")
    static T40 = new TypeContainer("T40")
    constructor(name){
        this.name = name;
    }
}

export class ContainerStatus{
    static Full = new ContainerStatus("Full")
    static Empty = new ContainerStatus("Empty")
    constructor(name){
        this.name = name;
    }
}

export class ContainerCategory{
    static Importation = new ContainerCategory("Importation");
    static Exportation = new ContainerCategory("Exportation");
    constructor(name){
        this.name = name;
    }
}

File: cool_load_backend/src/model/container.rs
use serde::{Deserialize, Serialize};
use rocket::data::FromData;
use super::client::Client;

#[derive(Serialize,Deserialize,Debug)]
pub struct Container{
    pub id: String,
    pub client: Client,
    pub type_container : TypeContainer,
    pub status : ContainerStatus,
    pub category : ContainerCategory 
}

#[derive(Serialize,Deserialize,Debug)]
pub enum TypeContainer{
    T20,
    T40
}


impl TypeContainer{
    pub fn from(txt : String) -> Self{
        if txt == "T20"{
            Self::T20
        }else{
            Self::T40
        } 
    }
}

#[derive(Serialize,Deserialize,Debug)]
pub enum ContainerStatus{
    Empty,
    Full
}

impl ContainerStatus{
    pub fn from(txt : String) -> Self{
        if txt == "Empty"{
            Self::Empty
        }else{
            Self::Full
        } 
    }
}

#[derive(Serialize,Deserialize,Debug)]
pub enum ContainerCategory{
    Importation,
    Exportation
}
impl ContainerCategory{
    pub fn from(txt : String) -> Self{
        if txt == "Importation"{
            Self::Importation
        }else{
            Self::Exportation
        } 
    }
}
